File: Selenium/Activities/Activity8.py
import time

from selenium import webdriver
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

from selenium.webdriver.common.by import By

serv = FirefoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=serv) as driver:
    driver.get("https://training-support.net/selenium/dynamic-controls")
    print(driver.title)
    wait = WebDriverWait(driver, 10)
    checkbox = driver.find_element(By.ID, "dynamicCheckbox")
    checkbox_toggle = driver.find_element(By.ID, "toggleCheckbox")

    checkbox_toggle.click()
    print("Checkbox hidden")
    wait.until(expected_conditions.invisibility_of_element(checkbox))
    checkbox_toggle.click()
    print("Checkbox visible")
    checkbox.click()
    print("Checkbox selected")
    driver.quit()

File: Selenium/Activities/Activity8.java
package Activities;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedCondition;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import java.time.Duration;
import static org.openqa.selenium.support.ui.ExpectedConditions.*;

public class Activity8 {
    public static void main(String[] args) throws InterruptedException
    {
        WebDriverManager.firefoxdriver().setup();
        WebDriver newDriver = new FirefoxDriver();
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE,"/dev/null");
        WebDriverWait wait = new WebDriverWait(newDriver, Duration.ofSeconds(10));
        newDriver.get("https://training-support.net/selenium/dynamic-controls");
        System.out.println("Title is:"+newDriver.getTitle());
        newDriver.findElement(By.id("toggleCheckbox")).click();
        Thread.sleep(2000);
        WebElement checkboxEle = newDriver.findElement(By.xpath("//input[@class='willDisappear']"));
        wait.until(ExpectedConditions.invisibilityOf(checkboxEle));
        System.out.println(checkboxEle.isDisplayed());
        newDriver.findElement(By.id("toggleCheckbox")).click();
        Thread.sleep(2000);
        wait.until(ExpectedConditions.elementToBeClickable(checkboxEle));
        System.out.println(checkboxEle.isDisplayed());
        checkboxEle.click();
        Thread.sleep(2000);
        newDriver.quit();
    }
}
